crates/gen-host-js/src/lib.rs

```diff
@@ -1151,12 +1193,16 @@ impl<'a> JsInterface<'a> {
                         }
                     }
                     TypeDefKind::Result(r) => {
-                        self.needs_ty_result = true;
-                        self.src.ts("Result<");
-                        self.print_optional_ty(r.ok.as_ref());
-                        self.src.ts(", ");
-                        self.print_optional_ty(r.err.as_ref());
-                        self.src.ts(">");
+                        if self.throw_ty_result {
+                            self.print_optional_ty(r.ok.as_ref());
+                        } else {
+                            self.needs_ty_result = true;
+                            self.src.ts("Result<");
+                            self.print_optional_ty(r.ok.as_ref());
+                            self.src.ts(", ");
+                            self.print_optional_ty(r.err.as_ref());
+                            self.src.ts(">");
+                        }
```

Comment:
I think this may not be correct for something like `-> list<result<...>>` since it won't print the `result`?

This is sort of similar to the above question about removing typescript since it doesn't seem to be tested and is falling behind.

Reply:
I'm using `throw_ty_result` as a single visitor marker more like an argument than a global state by immediately reverting its state after the call. See where this is being set. I'm open to actually treating it like an argument or better.

Reply:
Note that TypeScript is still being tested by the strong TSC checking in the tests.

Reply:
Okay, in this case I believe we do need a better way to pipe this low-level information through eg via some core ABI information. I'd appreciate some advice on how to do that.

Reply:
I just inlined the type logic explicitly for now, avoiding the need for `throw_ty_result` tracking.